src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3957,6 +3957,39 @@ protected Object stringByteIndex(
         }
     }
 
+    /** Search pattern in string starting at offset bytes backwards, and return a byte index or nil */
+    @Primitive(name = "string_byte_reverse_index", lowerFixnum = 3)
+    public abstract static class StringByteReverseIndexNode extends PrimitiveArrayArgumentsNode {
+        @Specialization
+        protected Object stringByteIndex(
+                Object rubyString, Object rubyPattern, RubyEncoding compatibleEncoding, int byteOffset,
+                @Cached RubyStringLibrary libString,
+                @Cached RubyStringLibrary libPattern,
+                @Cached TruffleString.LastByteIndexOfStringNode lastByteIndexOfStringNode,
+                @Cached ConditionProfile indexOutOfBoundsProfile,
+                @Cached ConditionProfile foundProfile) {
```

Comment:
This should use InlinedConditionProfile since https://github.com/oracle/truffleruby/commit/404a0864b7e53311f432136de65097e6a4ba4800 merged.

And @horakivo we should add a CheckStyle rule to catch this and other profiles, so we don't accidentally use `@Cached` but not inlined profiles.

Reply:
Done